### PR TITLE
Refuse to link live timelines into the forwards/backwards position when either is invalid

### PR DESCRIPTION
**Pull request by turt2live**
See https://github.com/vector-im/riot-web/issues/8593#issuecomment-478681816

Previously (https://github.com/matrix-org/matrix-js-sdk/pull/873) we allowed half-linking timelines to each other if they satisfy the conditions, however this appears to not be helping. Instead, it seems like the timelines are getting stuck in a position where one direction is spliced but the other is broken. To avoid this case, we'll just avoid splicing in both directions when one of the directions is invalid.

### CODE REVIEW

**Comment by turt2live** [comment]
It would be true only in the cases where the live timeline is being put in a position where it cannot be live (ie: A -> B -> C, where B would be the live timeline). The variables are already a bit confusing here, but `forwardIsLive` uses the inverse of `backwardsIsLive`, just like it was before. 

The problem this aims to solve is certainly vague, and it might very well indeed not fix anything. The concern is that by allowing 50% of a link to go through, we're actually making the problem worse. The theory is that by preventing the link altogether that we'll end up with a stable timeline which can be nicely spliced later on if needed.

**Comment by bwindels** [comment]
Ok, so the only actual change is the continue statement so setNeighbour ... doesn't get called for either timelines. :+1:

File: src/models/event-timeline-set.js [hunk 1]
```diff
@@ -413,26 +413,31 @@ EventTimelineSet.prototype.addEventsToTimeline = function(events, toStartOfTimel
...
+        if (backwardsIsLive || forwardsIsLive) {
             // The live timeline should never be spliced into a non-live position.
-            console.warn(
-                "Refusing to set our preceding timeline on a existingTimeLine " +
-                "as our timeline is live (" + timeline + ")",
-            );
-        } else {
-            existingTimeline.setNeighbouringTimeline(timeline, inverseDirection);
+            // We use independent logging to better discover the problem at a glance.
+            console.warn({backwardsIsLive, forwardsIsLive}); // debugging
```
Comment (bwindels) [inline src/models/event-timeline-set.js]: is this intended to stay in? Should be clear from seeing either of the below messages already?

Comment (turt2live) [reply src/models/event-timeline-set.js]: It's intended to stay in to make it slightly easier to troubleshoot from the logs without having to go hunting for messages in the code.

**Pull request merged**